src/Home/LocationActivationStatus.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { FunctionComponent } from "react"
+import { Alert, Linking } from "react-native"
+import { useNavigation } from "@react-navigation/native"
+import { useTranslation } from "react-i18next"
+
+import { useHasLocationRequirements } from "./useHasLocationRequirements"
+import { ActivationStatus } from "./ActivationStatus"
+import { HomeScreens } from "../navigation"
+
+export const LocationActivationStatus: FunctionComponent = () => {
+  const {
+    isLocationOn,
+    isLocationNeeded: showLocationStatus,
+  } = useHasLocationRequirements()
+  const navigation = useNavigation()
+  const { t } = useTranslation()
+
+  const handleOnPressFix = () => {
+    showFixLocationAlert()
+  }
+
+  const handleOnPressShowInfo = () => {
+    navigation.navigate(HomeScreens.LocationInfo)
+  }
+
+  const showFixLocationAlert = () => {
+    Alert.alert(
+      t("home.bluetooth.location_disabled_error_title"),
+      t("home.bluetooth.location_disabled_error_message"),
+      [
+        {
+          text: t("common.cancel"),
+          style: "cancel",
+        },
+        {
+          text: t("common.settings"),
+          onPress: () => Linking.openSettings(),
+        },
+      ],
+    )
+  }
+
+  return (
+    <>
+      {showLocationStatus && (
```

Comment:
I'm not opposed to moving this logic back up per se, but I think this is part of why the Home component became so unreadable. It was using a _lot_ of state to determine which children to render, which  meant extra variable declarations and many ternaries and conditionals in the `return`. This became even harder to follow because the variable definitions were really far from where those variables were used.

In this case, I might prefer to have the ActivationStatus and the ShareLink be responsible for reading from context and displaying themselves correctly. I think it's easier to read because these components are much simpler, so there is only one conditional in the return and it is directly related to the component's display logic. What do you think?

cc @johnschoeman because we made a similar decision in the ShareLink

Reply:
I'm sorry I meant a little higher in this exact component, not back up to the parent one, 💯 agree with you on this reasoning, moving it here and making this component responsible for its own logic is the perfect move. Just thought that since now is more compact and nice to understand doing:

``` typescript
if(!showLocationStatus) {
  return null
}
```

a little higher would not be terribly harder to read. But it's definitely not a big deal.

Reply:
Oooh I got it. That makes sense, thanks for clarifying. I will make this change!